File: test/st/output/plc.cpp
#include "imperium.h"
#include "modbus.h"
class Timer {//FUNCTION_BLOCK:Timer
public:
  bool Start;
  bool Done;
  int16_t Counter;
  void operator()() {
    Counter = Counter + 1;
  }
};

void PLC_PROG() { //PROGRAM:PLC_PROG
auto T1;
uint64_t IN;
IN = % I0001 . 0;
// uncompilable statement {"type":"IF","condition":"T1","thenBlock":null,"elseIfBlocks":[],"elseBlock":null}
0 = 1;
Done = true THEN T1 . Start : == false;
0 = 0;
}


int main() {
  uint cycleCount = 0;
  while (true) {
    gatherInputs();
    PLC_PROG();

    handleOutputs();
    std::this_thread::sleep_for(std::chrono::milliseconds(1));
    cycleCount++;
    if(cycleCount >= 604800000){
        cycleCount = 0;
    }
   }
  return 0;
}

File: test/st/output/plc.js
import {
        readBit, writeBit, readByte, writeByte, readWord, writeWord, readDWord, writeDWord,
        getBit, setBit, resolve, newStatic, RefVar, superviseIO, mapIO, createReference,
        TON, TOF, TP, R_TRIG, F_TRIG, CTU, CTD, CTUD,
        AND, OR, XOR, NOR, NAND, NOT, ASSIGNMENT,
        EQ, NE, LT, GT, GE, LE,
        MOVE, SEL, MUX, MIN, MAX, LIMIT
} from "./imperium.js";
export class Timer { // FUNCTION_BLOCK:Timer
  constructor() {
    this.Start = null;
    this.Done = null;
    this.Counter = 0;
  }
  call() {
    this.Counter = resolve(this.Counter) + 1;
  }
}
export function PLC_PROG() { // PROGRAM:PLC_PROG
let T1 = newStatic("T1", Timer);
let IN;
false
PLC_PROG
T1.call();
IN = readBit("%I0001.0");
if (resolve(T1.Start) == true && ! resolve(T1.Done)) {
  writeBit("%Q0001.0", 1);
}
else if (resolve(T1.Done) == true) {
  T1.Start = false;
  writeBit("%Q0001.0", 0);
}
}

export function setup(){
    
    console.log("ImperiumPLC is running!");
}

export function run(){
    setInterval(superviseIO, 1); 
    setInterval(() => {
PLC_PROG();
}, 100);
    
}

setup();
run();